### PR TITLE
Simplify some code.

### PR DESCRIPTION
**Pull request by bangerth**
Follow-up to #10716. @peterrum -- FYI.

/rebuild

### CODE REVIEW

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 1]
```diff
@@ -828,37 +828,25 @@ namespace internal
...
-
             unsigned int index = 0;
 
             // 1) VERTEX dofs
-            for (const auto vertex : accessor.vertex_indices())
-              {
-                (void)vertex;
-                index += dofs_per_vertex;
-              }
+            index += dofs_per_vertex * accessor.vertex_indices().size();
```
Comment (peterrum) [inline include/deal.II/dofs/dof_accessor.templates.h:834]:
```suggestion
            index += dofs_per_vertex * accessor.n_vertices;
```

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 2]
```diff
@@ -828,37 +828,25 @@ namespace internal
...
 
             // 2) LINE dofs
             if (structdim == 2 || structdim == 3)
-              for (const auto line : accessor.line_indices())
-                {
-                  (void)line;
-                  index += dofs_per_line;
-                }
+              index += dofs_per_line * accessor.line_indices().size();
```
Comment (peterrum) [inline include/deal.II/dofs/dof_accessor.templates.h:838]:
```suggestion
              index += dofs_per_line * accessor.n_lines();
```

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 3]
```diff
@@ -828,37 +828,25 @@ namespace internal
...
 
             // 3) FACE dofs
             if (structdim == 3)
-              for (const auto face : accessor.face_indices())
-                {
-                  (void)face;
-                  index += dofs_per_quad;
-                }
+              index += dofs_per_quad * accessor.face_indices().size();
```
Comment (peterrum) [inline include/deal.II/dofs/dof_accessor.templates.h:842]:
```suggestion
              index += dofs_per_quad * accessor.n_faces();
```

**Comment by bangerth** [comment]
Comments addressed.

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 4]
```diff
@@ -825,37 +825,25 @@ namespace internal
...
-
             unsigned int index = 0;
 
             // 1) VERTEX dofs
-            for (const auto vertex : accessor.vertex_indices())
-              {
-                (void)vertex;
-                index += dofs_per_vertex;
-              }
+            index += dofs_per_vertex * accessor.n_vertices;
```
Comment (masterleinad) [inline include/deal.II/dofs/dof_accessor.templates.h:831]:
```suggestion
            index += dofs_per_vertex * accessor.n_vertices();
```

**Comment by bangerth** [comment]
Ah, don't trust suggested edits :-) I pushed a version that now compiles cleanly locally.

**Pull request merged**